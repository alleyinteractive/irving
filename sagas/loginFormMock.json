{
  "tenantId": "technologyreview-staging",
  "subTenantId": "technologyreview-staging",
  "slug": "login",
  "title": "Login",
  "registration": true,
  "allowSocialSignIn": false,
  "providers": [],
  "fields": [
    {
      "order": 1,
      "placeholder": "Enter your email address",
      "required": true,
      "slug": "email-address",
      "default-value": "",
      "display-condition": "always"
    },
    {
      "order": 2,
      "placeholder": "Enter your password",
      "required": true,
      "slug": "password",
      "default-value": "",
      "display-condition": "always"
    }
  ],
  "allowAccountRegistration": false,
  "css": ".blaize-theme-default.blaize-form {\n margin: 20px auto;\n padding: 20px;\n border-radius: 3px;\n background-color: #fff;\n}\n.blaize-theme-default.blaize-form fieldset {\n border: none;\n padding: 0;\n margin: 0;\n}\n.blaize-theme-default.blaize-form .blaize-title {\n margin-top: 0;\n}\n.blaize-theme-default.blaize-form .blaize-error {\n display: none;\n color: #e12d2d;\n}\n.blaize-theme-default.blaize-form .blaize-success {\n display: none;\n margin: 1em 0;\n background-color: #eef5ee;\n border: 1px solid #638463;\n border-radius: 2px;\n color: #638463;\n padding: 1em;\n font-size: 0.9rem;\n}\n.blaize-theme-default.blaize-form input,\n.blaize-theme-default.blaize-form select {\n width: 100%;\n height: 42px;\n display: block;\n margin-bottom: 15px;\n border: 1px solid #e1e3e8;\n border-radius: 3px;\n background-color: white;\n}\n.blaize-theme-default.blaize-form .blaize-submit {\n display: block;\n width: 180px;\n margin: 10px auto 0;\n height: 42px;\n font-weight: bold;\n background-color: #4697ce;\n color: #fff;\n border-radius: 3px;\n}\n.blaize-theme-default.blaize-form input {\n padding: 5px 10px;\n}\n.blaize-theme-default.blaize-form input[type=\"checkbox\"] {\n display: inline-block;\n width: auto;\n height: auto;\n margin: 0 5px;\n}\n.blaize-theme-default.blaize-form .blaize-social-sign-in-footer {\n text-align: center;\n overflow: hidden;\n margin: 15px 0 10px 0;\n font-weight: bold;\n}\n.blaize-theme-default.blaize-form .blaize-social-sign-in-footer:before,\n.blaize-theme-default.blaize-form .blaize-social-sign-in-footer:after {\n border-bottom: 1px solid #e1e3e8;\n content: \"\";\n display: inline-block;\n width: 50%;\n margin: 0 6px 0 -55%;\n vertical-align: middle;\n}\n.blaize-theme-default.blaize-form .blaize-social-sign-in-footer:after {\n margin: 0 -55% 0 6px;\n}\n.blaize-theme-default.blaize-form.blaize-form-register,\n.blaize-theme-default.blaize-form.blaize-form-login,\n.blaize-theme-default.blaize-form.blaize-form-reset-password {\n max-width: 420px;\n}\n.blaize-theme-default.blaize-form.blaize-form-login,\n.blaize-theme-default.blaize-form.blaize-form-reset-password {\n display: none;\n}\n.blaize-theme-default.blaize-form.blaize-form-register .blaize-submit,\n.blaize-theme-default.blaize-form.blaize-form-login .blaize-submit {\n width: 100%;\n margin-bottom: 10px;\n}",
  "action": "",
  "theme": "default",
  "internal-description": "",
  "public-description": ""
}