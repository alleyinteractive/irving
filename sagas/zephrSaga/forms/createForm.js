import React from 'react';

/**
 * Construct a form via a response from the Zephr API.
 *
 * @param {object} formJSON Stringified JSON input from Zephr API.
 *
 * @returns {array} Array of React elements to use in the Zephr form.
 */
export default function createForm(formJSON) {
  const {
    input: {
      slug,
      fields,
      registration,
    },
    submitText,
  } = formJSON;

  // Map the fields to React components.
  let components = fields.map((field) => {
    const {
      slug: id,
      placeholder,
      required,
      'default-value': defaultValue,
    } = field;

    let type = '';
    switch (id) {
      case 'email-address':
        type = 'email';
        break;
      default:
        type = 'text';
    }

    let props = {
      key: id,
      id,
      className: `zephr-input-${id}`,
      type,
      placeholder,
      required,
      defaultValue,
    };

    if ('email-address' === id) {
      props = { ...props, autoComplete: 'username' };
    }

    return React.createElement('input', props, null);
  });

  if (true === registration) {
    components = [...components, ...generatePasswordFields(slug)];
  }

  const buttonProps = {
    key: 'zephr-submit-button',
    type: 'submit',
    value: submitText,
  };

  // Create the submit button.
  const submitButton = React.createElement('input', buttonProps, null);

  // Append the submit button to the end of the array.
  components = [...components, submitButton];

  // Return the components array.
  return components;
}

function generatePasswordFields(slug) {
  const id = 'registration' === slug ? 'new-password' : 'current-password';
  const placeholder =
    'registration' === slug ?
      'Create a password for your account' :
      'Enter your password';

  const props = {
    key: id,
    id,
    className: `zephr-input-${id}`,
    type: 'password',
    placeholder,
    required: true,
    defaultValue: '',
  };

  const fields = [
    React.createElement('input', props, null),
  ];

  if ('registration' === slug) {
    const verifyId = 'verify-password';

    const verifyProps = {
      key: verifyId,
      id: verifyId,
      className: `zephr-input-${verifyId}`,
      type: 'password',
      placeholder: 'Confirm your password',
      required: true,
      defaultValue: '',
    };

    fields.push(React.createElement('input', verifyProps, null));
  }

  return fields;
}
