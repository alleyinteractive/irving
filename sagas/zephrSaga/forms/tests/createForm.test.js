import createForm from '../createForm';
import loginFormMock from './loginMock.json';
import registrationFormMock from './registrationMock.json';

describe('zephr saga => createForm HOC', () => {
  it('should render a login form of the shape', () => {
    const form = createForm(loginFormMock);
    expect(form).toMatchSnapshot();
  });

  it('should render a registration form of the shape', () => {
    const form = createForm(registrationFormMock);
    expect(form).toMatchSnapshot();
  });
});
