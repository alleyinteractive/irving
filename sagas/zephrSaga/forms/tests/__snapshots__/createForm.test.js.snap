// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zephr saga => createForm HOC should render a login form of the shape 1`] = `
Array [
  <FormInput
    autoComplete="username"
    className="zephr-input-email-address"
    defaultValue=""
    id="email-address"
    placeholder="Enter your email address"
    required={false}
    type="email"
  />,
  <FormInput
    autoComplete="current-password"
    className="zephr-input-current-password"
    defaultValue=""
    id="current-password"
    placeholder="Enter your password"
    required={true}
    type="password"
  />,
  <input
    type="submit"
    value="Login"
  />,
]
`;

exports[`zephr saga => createForm HOC should render a registration form of the shape 1`] = `
Array [
  <FormInput
    autoComplete="username"
    className="zephr-input-email-address"
    defaultValue=""
    id="email-address"
    placeholder="Enter your email address"
    required={false}
    type="email"
  />,
  <FormInput
    autoComplete=""
    className="zephr-input-full-name"
    defaultValue=""
    id="full-name"
    placeholder="Enter your full name"
    required={true}
    type="text"
  />,
  <FormInput
    autoComplete="new-password"
    className="zephr-input-new-password"
    defaultValue=""
    id="new-password"
    placeholder="Create a password for your account"
    required={true}
    type="password"
  />,
  <FormInput
    autoComplete="new-password"
    className="zephr-input-verify-password"
    defaultValue=""
    id="verify-password"
    placeholder="Confirm your password"
    required={true}
    type="password"
  />,
  <TermsCheckbox
    id="terms-checkbox"
  />,
  <input
    type="submit"
    value="Create an account"
  />,
]
`;
